### PR TITLE
add default NAStrategy to models

### LINKED ISSUE #178: Add default NAStrategy to models
Currently, all the models don't have a default NAStrategy. If a user runs any model on data containing nans, then all the gradients would explode.

### PR DESCRIPTION
**Pull request by yiweny**
Fixes #178

### CODE REVIEW

File: test/nn/models/test_compile.py [hunk 1]
```diff
@@ -54,7 +54,7 @@
             Trompt,
             dict(channels=8, num_prompts=2),
             None,
-            6,
+            10,
```
Comment (akihironitta) [inline test/nn/models/test_compile.py:57]: For the note, the reason we increase the count is:
```
    Reason: Dynamic control flow is not supported at the moment. Please use functorch.experimental.control_flow.cond to explicitly capture the control flow
    User Stack:
      <FrameSummary file /home/akihiro/work/github.com/pyg-team/pytorch-frame/torch_frame/nn/encoder/stype_encoder.py, line 90 in forward>
      <FrameSummary file /home/akihiro/work/github.com/pyg-team/pytorch-frame/torch_frame/nn/encoder/stype_encoder.py, line 140 in na_forward>
```

File: torch_frame/nn/models/trompt.py [hunk 2]
```diff
@@ -75,10 +76,13 @@ def __init__(
...
                         stype.categorical:
-                        EmbeddingEncoder(post_module=LayerNorm(channels)),
+                        EmbeddingEncoder(post_module=LayerNorm(channels),
+                                         na_strategy=NAStrategy.MOST_FREQUENT),
                         stype.numerical:
-                        LinearEncoder(post_module=Sequential(
-                            ReLU(), LayerNorm(channels))),
+                        LinearEncoder(
+                            post_module=Sequential(ReLU(),
+                                                   LayerNorm(channels)),
```
Comment (rusty1s) [inline torch_frame/nn/models/trompt.py:84]:
```suggestion
                                                   LayerNorm(channels), ),
```

**Review by rusty1s — APPROVED** [review]
